docs/flow.md

```diff
@@ -1777,6 +1779,127 @@ as cancellation and structured concurrency serve this purpose.
 Note that [launchIn] also returns a [Job], which can be used to [cancel][Job.cancel] the corresponding flow collection
 coroutine only without cancelling the whole scope or to [join][Job.join] it.
  
+### Flow cancellation checks
+
+For convenience, the [flow] builder performs additional [ensureActive] checks for cancellation on each emitted value. 
+It means that a busy loop emitting from a `flow { ... }` is cancellable:
+ 
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+
+//sampleStart           
+fun foo(): Flow<Int> = flow { 
```

Comment:
[Nit] For guides it would be nice to stick with a meaningful naming (e.g. `simpleFlow`) instead of just `foo`

Reply:
I agree, `foo` and `bar` are confusing to many people.
I like to use "some" as a prefix, and continue with what it is for extra clarity.
Examples: `someFlow`, `SomeClass`, `someFunction()`, `SomeType.someExtensionProperty`.

There's some alternatives to "some" like "aRandom", or suffixing with number literals when there's two or more similarly kinded symbols (better than foo and bar IMHO).

Reply:
I'll follow up with `foo` banishment in a separate change.